### PR TITLE
🩹 Add a command to patch connector parameters

### PR DESCRIPTION
**Pull request by karim-elngr**
#36

### CODE REVIEW

File: src/main/java/dev/morling/kccli/command/PatchConnectorCommand.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        String connectorParametersString = new ObjectMapper().writeValueAsString(connectorParameters);
+        kafkaConnectApi.updateConnector(name, connectorParametersString);
+
+        DescribeConnectorCommand describeConnectorCommand = new DescribeConnectorCommand();
+        describeConnectorCommand.context = context;
+        describeConnectorCommand.name = name;
+        describeConnectorCommand.includeTasksConfig = false;
+
+        return describeConnectorCommand.call();
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/kccli/command/PatchConnectorCommand.java:64]: Perhaps print out a statement here such as "New connector configuration:"?

**Review by gunnarmorling — APPROVED** [review]
Very nice! One small suggestion inline. How does unsetting a connector option work? Can I do `-p foo=null`? Or maybe we should have both `-s` ("set") and `-r` ("remove")? WDYT?

Besides that, could add a completion which returns the already existing options (I've just pushed a commit which makes completion handing quite a bit simpler)? Then users will have <TAB> completion when updating an option, and will need to fully type the name only for adding a new connector config option.

File: src/main/java/dev/morling/kccli/command/PatchConnectorCommand.java (hunk 1 above)
Comment (karim-elngr) [reply src/main/java/dev/morling/kccli/command/PatchConnectorCommand.java:64]: Yes, let me add it

**Comment by karim-elngr** [comment]
> Very nice! One small suggestion inline. How does unsetting a connector option work? Can I do `-p foo=null`? Or maybe we should have both `-s` ("set") and `-r` ("remove")? WDYT?

I thought so as well, we can introduce the null but we will have to parse the `"null"` string and remove the config from the map, alternatives I like the 2nd approach but perhaps we can omit the `-s` and default to adding a config unless `-r` is specified, WDYT?

**Comment by karim-elngr** [comment]
> Besides that, could add a completion which returns the already existing options (I've just pushed a commit which makes completion handing quite a bit simpler)? Then users will have completion when updating an option, and will need to fully type the name only for adding a new connector config option.

I could not find any hints as to how to make auto-completion work based on a previous argument since we need to load configuration properties based on the connector name

**Comment by gunnarmorling** [comment]
> I think I prefer the 2nd option because it seems less ambiguous, WDYT?

+1

> I could not find any hints as to how to make auto-completion work based on a previous argument since we need to load configuration properties based on the connector name

[This article](https://iridakos.com/programming/2018/03/01/bash-programmable-completion-tutorial) could be helpful; e.g. see `COMP_WORDS` and the related env vars which provide access to details of the current command line. You'd then have to add a parameter to the definition of the command invoked by the completion.

**Comment by karim-elngr** [comment]
Hello @gunnarmorling I did not yet get to finish reading it, but I do not mind merging the PR and add a follow up to add the auto completion

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Ok, cool. PR merged. Thx a lot, @karim-elngr!

**Pull request merged**